### PR TITLE
Adding support for BYO OpenAI to Standard deployment

### PR DESCRIPTION
**Pull request by hatboyzero**
# Adding support for BYO OpenAI to Standard deployment

## The issue or feature being addressed

Fixes #17318

## Details on the issue fix or feature implementation

Updates bicep templates and automation scripts to add support for BYO OpenAI.

## Confirm the following

- [ ]  I started this PR by branching from the head of the default branch
- [ ]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [ ]  I have included inline docs for my changes, where applicable
- [ ]  I have successfully run a local build
- [ ]  I have provided the required update scripts, where applicable
- [ ]  I have updated relevant docs, where applicable

### CODE REVIEW

File: deploy/standard/bicep/main.bicep [hunk 1]
```diff
@@ -10,6 +10,10 @@ param createVpnGateway bool = false
 param environmentName string
 param externalDnsResourceGroupName string = ''
 param externalNetworkingResourceGroupName string = ''
+param deployOpenAi bool = true
```
Comment (jamesrcounts) [inline deploy/standard/bicep/main.bicep:13]: This should be a var and not part of the public interface

File: deploy/standard/bicep/psakefile.ps1 [hunk 2]
```diff
@@ -77,6 +77,22 @@ task Main -depends Configuration {
             type  = "string"
             value = $script:environment
         }
+        deployOpenAi                = @{
```
Comment (jamesrcounts) [inline deploy/standard/bicep/psakefile.ps1:80]: see the pattern for external DNS and external network below

**Review by jamesrcounts — CHANGES_REQUESTED** [review]

File: deploy/standard/bicep/main.bicep (hunk 1 above)
Comment (hatboyzero) [reply deploy/standard/bicep/main.bicep]: Updated.

File: deploy/standard/bicep/psakefile.ps1 (hunk 2 above)
Comment (hatboyzero) [reply deploy/standard/bicep/psakefile.ps1]: Updated.

**Review by jamesrcounts — APPROVED** [review]

**Review by jamesrcounts — APPROVED** [review]

**Review by ciprianjichici — APPROVED** [review]

**Pull request merged**